src/swell/utilities/slurm.py

```diff
@@ -36,7 +45,13 @@ def prepare_scheduling_dict(
     # ----------------------------------------------
     # NOTE: Separate function to allow it to be mocked in unit tests.
     # See https://github.com/GEOS-ESM/swell/issues/351
-    user_globals = slurm_global_defaults(logger)
+    # user_globals = slurm_global_defaults(logger)
+
+    # Check if platform contains Linux-5.14.21, which indicates platform is SLES15
+    if 'Linux-5.14.21' in pltfrm.platform():
+        current_platform = "nccs_discover_sles15"
+        assert platform == current_platform, \
+            f"Inconsistent platfrom choice: {platform} vs. {current_platform}"
```

Comment:
Suggest a more informative error message:

```suggestion
        assert platform == "nccs_discover_sles15", (
            "'Linux-5.14.21' detected, which implies platform 'nccs_discover_sles15. " +
            f"That is inconsistent with user-specified platform '{platform}'."
        )
```